test/e2e/independent_customroles_test.go

```diff
@@ -134,9 +134,11 @@ var _ = Describe("Migrate one CustomRole from AtlasProject to AtlasCustomRole re
 		})
 
 		By("Enabled reconciliation for AtlasProject", func() {
-			Expect(testData.K8SClient.Get(testData.Context, client.ObjectKeyFromObject(testData.Project), testData.Project)).Should(Succeed())
-			testData.Project.Annotations = map[string]string{}
-			Expect(testData.K8SClient.Update(testData.Context, testData.Project)).To(Succeed())
+			Eventually(func(g Gomega) {
+				g.Expect(testData.K8SClient.Get(testData.Context, client.ObjectKeyFromObject(testData.Project), testData.Project)).Should(Succeed())
+				testData.Project.Annotations = map[string]string{}
+				g.Expect(testData.K8SClient.Update(testData.Context, testData.Project)).To(Succeed())
+			}).WithTimeout(1 * time.Minute).WithPolling(5 * time.Second).Should(Succeed())
```

Comment:
Still I think @s-urbaniak had an update implementation to precisely avoid race condition conflicts on updated

Reply:
yep, see [RetryUpdateOnConflict](https://github.com/mongodb/mongodb-atlas-kubernetes/blob/b6bbb6db6ce1c81b546f30babea890f5e5bc1554/test/helper/retry/retry.go#L17)

And also see [sample usage](https://github.com/mongodb/mongodb-atlas-kubernetes/blob/b6bbb6db6ce1c81b546f30babea890f5e5bc1554/test/e2e/backup_compliance_test.go#L145)

Reply:
yeah, let's use the canonical one.

Reply:
Why is it better? I prefer the explicit way of doing such checks to be honest.

Reply:
This is not only also explicit, but less error prone, just a single function call:
```go
			_, err := akoretry.RetryUpdateOnConflict(
				ctx,
				testData.K8SClient,
				client.ObjectKeyFromObject(...),
				func(...) {...})
			Expect(err).To(BeNil())
```

Reply:
@igor-karpukhin i highly suggest using the central function, it is a small wrapper around the central client-go `RetryUpdateOnConflictFunction`, see https://github.com/kubernetes/client-go/blob/master/util/retry/util.go#L68-L101.

This function is battle-tested throughout Kubernetes and used in many operators, it includes backoff and considers only Conflict errors which both things are not handled by your function.